### ISSUE TITLE
Keeping PNG transparency

### ISSUE THREAD

**Issue by halexic** [LazZiya/ImageResize#4]
Is it possible to keep PNG transparency after resizing/scaling and saving?

**Comment by halexic** [comment]
Hi there,

I found the improvement how to scale and save transparent PNG images by keeping the transparency. 

In ImageResize.cs replace this method "Resize" like this:

```
/// <summary>
/// Specify custom resize options
/// </summary>
/// <param name="img">the image to resize</param>
/// <param name="source">The coordinates to read as source from the image, 
/// can be the whole image or part of it</param>
/// <param name="target">The coordinates of the target image size</param>
/// <param name="ops">Graphic options <see cref="GraphicOptions"/></param>
/// <returns></returns>
public static Image Resize(this Image img, Rectangle source, Rectangle target, GraphicOptions ops)
{
  var pixF = ImageColorFormats.GetColorFormat((Bitmap)img) == ImageColorFormat.Cmyk || ImageColorFormats.GetColorFormat((Bitmap)img) == ImageColorFormat.Rgb
      ? PixelFormat.Format32bppArgb
      : img.PixelFormat;
  if (img.RawFormat.Guid == ImageFormat.Png.Guid)
  {
    pixF = PixelFormat.Format32bppArgb;
  }

  using (Bitmap outputImage = new Bitmap(target.Width, target.Height, pixF))
  {
    outputImage.SetResolution(img.HorizontalResolution, img.VerticalResolution);

    using (var graphics = Graphics.FromImage(outputImage))
    {
      Rectangle destRectangle = new Rectangle(0, 0, target.Width, target.Height); //Actual pixels
      graphics.SmoothingMode = ops.SmoothingMode;
      graphics.InterpolationMode = ops.InterpolationMode;
      graphics.PixelOffsetMode = ops.PixelOffsetMode;
      graphics.CompositingQuality = ops.CompositingQuality;
      graphics.CompositingMode = ops.CompositingMode;
      graphics.PageUnit = ops.PageUnit;
      if (img.RawFormat.Guid == ImageFormat.Png.Guid)
      {
        graphics.Clear(Color.Transparent);
      }
      graphics.DrawImage(
          img,
          target,
          source,
          ops.PageUnit);
    }

    if (img.RawFormat.Guid == ImageFormat.Png.Guid)
    {
      using (MemoryStream memoryStream = new MemoryStream())
      {
        // in this case i needed to return a webimage,
        //but you could simply save it to disk, or return a byte array.
        //Again, be sure to save in a format that supports transparency,
        //such as Png does.
        outputImage.Save(memoryStream, ImageFormat.Png);
        return new Bitmap(memoryStream);
      }
    }
    else
    {
      return Image.FromHbitmap(outputImage.GetHbitmap());
    }        
  }
}
```

**Comment by LazZiya** [comment]
Hi @halexic ,

Thanks for creating this issue, I don't know how I skipped this before! 😅

I will check your code and update the package accordingly

**Comment by LazZiya** [comment]
With reference to your code, only saving to a memorystream solved the issue as below:

````cs
using (var ms = new MemoryStream())
{
    outputImage.Save(ms, img.RawFormat);
    return Image.FromStream(ms);
}
```` 

Fixed with version [3.0.1](https://github.com/LazZiya/ImageResize/releases/tag/v3.0.1)

**Comment by halexic** [comment]
Thanks for the optimization.

Best
Alex
